src/db/postgres/lrsql/postgres/sql/query.sql

```diff
@@ -351,3 +351,79 @@ WHERE id > :since-id
   AND id <= :until-id
 GROUP BY stored_time
 ORDER BY stored_time ASC;
+
+/* Statement Reactions */
+
+-- :snip snip-json-extract
+json_extract_path_text(:i:col, :v*:path):::sql:type
+
+-- :snip snip-jsonb-extract
+-- :doc essentially identical alternate for snip-json-extract for jsonb mode
+jsonb_extract_path_text(:i:col, :v*:path):::sql:type
+
+-- :snip snip-val
+:v:val
+
+-- :snip snip-col
+:i:col
+
+-- :snip snip-clause
+:snip:left :sql:op :snip:right
+
+-- :snip snip-and
+--~ (str "(" (apply str (interpose " AND " (map-indexed (fn [idx _] (str ":snip:clauses." idx)) (:clauses params)))) ")")
+
+-- :snip snip-or
+--~ (str "(" (apply str (interpose " OR " (map-indexed (fn [idx _] (str ":snip:clauses." idx)) (:clauses params)))) ")")
+
+-- :snip snip-not
+(NOT :snip:clause)
+
+-- :snip snip-contains-json
+-- :doc Does the json at col and path contain the given value? A special case with differing structure across backends
+(SELECT TRUE FROM json_array_elements_text(json_extract_path(:i:col, :v*:path)) WHERE value:::sql:type = :snip:right)
+
+-- :snip snip-contains-jsonb
+-- :doc Does the jsonb at col and path contain the given value? A special case with differing structure across backends
+(SELECT TRUE FROM jsonb_array_elements_text(jsonb_extract_path(:i:col, :v*:path)) WHERE value:::sql:type = :snip:right)
```

Comment:
I would definitely break these up into multiple lines, now that nested parens are involved.

Reply:
With the exception of the outer `snip-query-reaction` I've tried to keep the snippets as free from multiline formatting as possible. It is easier for me to debug the sqlvec that is produced by the dynamic query. I agree that it is hard to read on one line but not excessively so.

Reply:
I guess that's the difference between someone who wrote the code versus someone who's reading the code. I guess we'll agree to disagree on this then.

Reply:
No I'm saying there is an actual reason to not put formatting in here. If you need to read an SQLvec the extra spaces make it very hard to read.